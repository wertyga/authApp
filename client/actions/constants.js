export const SET_NEW_USER = 'SET_NEW_USER';