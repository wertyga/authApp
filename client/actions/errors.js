export const ERROR = 'ERROR';

export function globalError(err) {
    return {
        type: ERROR,
        error: err
    }
};