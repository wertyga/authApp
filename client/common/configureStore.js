import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from '../reducers/rootReducer';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

let store;
const dev = process.env.NODE_ENV === 'development';
if(dev) {
    store = createStore(
        rootReducer,
        composeWithDevTools(
            applyMiddleware(thunk)
        )
    );
} else {
    store = createStore(
        rootReducer,
        applyMiddleware(thunk)
    );
};

export default store;