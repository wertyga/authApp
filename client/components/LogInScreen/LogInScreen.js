import axios from 'axios';

import inputsValidation from '../../../server/middlewares/inputsValidation';

import Loading from '../../common/components/Loading';

import './LogInScreen.sass';

export class LogInScreen extends React.Component {
    constructor() {
        super();

        this.state = {
            username: '',
            password: '',
            errors: {},
            loading: false
        };
    };

    onChange = e => {
        const name = e.target.name;
        this.setState({
            [name]: e.target.value,
            errors: {
                ...this.state.errors,
                [name]: ''
            }
        });
    };

    Submit = e => {

        const sendObject = {
            username: this.state.username,
            password: this.state.password
        };
        const { isValid, errors } = inputsValidation(sendObject);

        if(isValid) {
            this.setState({ loading: true, errors: {}});
            const login = e.target.getAttribute('id') === 'login';
            const url = login ? '/auth/login' : '/auth/sign-up';

            return axios.post(url, { ...sendObject })
                .then(res => this.props.history.push(`/user/${res.data.id}`))
                .catch(err => {
                    err = err.response ? err.response.data.errors : { globalError: err.message };
                    this.setState({
                        errors: err,
                        loading: false
                    });
                })

        } else {
            this.setState({
                errors
            })
        }
    };

    render() {
        return (
            <div className="LogInScreen">
                <div className="form">
                    {this.state.errors.globalError && <div className="error">{this.state.errors.globalError}</div>}
                    {this.state.loading  && <Loading />}
                    <label htmlFor="usernameInput">Username:</label>
                    <input type="text" name="username" placeholder="Username..."
                           value={this.state.username} onChange={this.onChange} id="usernameInput"
                           disabled={this.state.loading}
                    />
                    {this.state.errors.username && <div className="error">{this.state.errors.username}</div>}

                    <label htmlFor="passwordInput">Password:</label>
                    <input type="text" name="password" placeholder="Password..."
                           value={this.state.password} onChange={this.onChange} id="passwordInput"
                           disabled={this.state.loading}
                    />
                    {this.state.errors.password && <div className="error">{this.state.errors.password}</div>}

                    <button className="primary"
                            disabled={this.state.loading || !this.state.username || !this.state.password}
                            onClick={this.Submit}
                            id="login"
                    >
                        Login
                    </button>
                    <button id="registration"
                            className="primary"
                            disabled={this.state.loading}
                            onClick={this.Submit}
                    >
                        Registration
                    </button>
                </div>
            </div>
        );
    };
};

export default LogInScreen;
