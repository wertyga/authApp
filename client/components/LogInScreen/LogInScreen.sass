.LogInScreen
  height: 100vh
  width: 100%
  padding-top: 20%
  //display: flex
  //justify-content: center
  .form
    display: flex
    flex-direction: column
    justify-content: center
    width: 50%
    margin: 0 auto
    label
      margin-top: 20px
      margin-bottom: 5px
      font-weight: bold
    button:first-of-type
      margin-top: 40px