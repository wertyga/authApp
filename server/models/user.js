import mongoose from 'mongoose';

const UserSchema = mongoose.Schema({
    username: String,
    password: String
});

export default mongoose.model('user', UserSchema);