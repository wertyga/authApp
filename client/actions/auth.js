import axios from 'axios';

import * as consts from './constants';

export function getUserData(id) {
    return dispatch => {
        return axios.get(`/user/${id}`)
            .then(res => dispatch(setUser(res.data.user)))
            .catch(err => console.log(err))
    };
};
export function setUser(user) {
    return {
        type: consts.SET_NEW_USER,
        user
    }
};