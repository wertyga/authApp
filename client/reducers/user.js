import { SET_NEW_USER } from '../actions/constants';

export const initialState = {};

export default function user(state = initialState, action = {}) {
    switch(action.type) {
        case SET_NEW_USER:
            return action.user;

        default:
            return state;
    };
};