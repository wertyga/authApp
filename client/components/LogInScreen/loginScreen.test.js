import { LogInScreen } from './LogInScreen';
import ConnectLogScreen from './LogInScreen';
import { loginUser, registration } from '../../actions/auth';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock from 'fetch-mock';

import { initialState } from '../../reducers/user';

const mockStore = configureStore([thunk]);

describe('#LoginScreen component', () => {
    it('Check initial state', done => {
        const wrapper = shallow(<LogInScreen />);

        expect(wrapper.state().username).toEqual('');
        expect(wrapper.state().password).toEqual('');
        expect(wrapper.state().errors).toEqual({});
        expect(wrapper.state().loading).toEqual(false);

        done()
    });

    it('Check error input fields, when require fields empty', done => {
        const wrapper = shallow(<LogInScreen />);

        wrapper.find('#login').simulate('click', {preventDefault: () => {}});
        expect(typeof wrapper.state().errors).toEqual('object'); 
        expect(wrapper.state().errors).toHaveProperty('username');
        expect(wrapper.state().errors).toHaveProperty('password');

        wrapper.find('#usernameInput').simulate('change', {
            target: {
                name: 'username',
                value: 'my username'
            }
        });
        expect(wrapper.state().username).toEqual('my username');
        wrapper.find('#login').simulate('click', {preventDefault: () => {}});
        expect(wrapper.state().errors).toHaveProperty('password');

        wrapper.find('#passwordInput').simulate('change', {
            target: {
                name: 'password',
                value: 'my password'
            }
        });
        wrapper.setState({ username: ''})
        expect(wrapper.state().password).toBe('my password');
        wrapper.find('#login').simulate('click', {preventDefault: () => {}});
        expect(wrapper.state().errors).toHaveProperty('username');

        done();
    });

    it("Check that component didn't change while require fields are empty", done => {
        const wrapper = shallow(<LogInScreen />);
        const tree = renderer.create(<LogInScreen />).toJSON();
        wrapper.find('#login').simulate('click', {preventDefault: () => {}});

        expect(tree).toMatchSnapshot();

        done();
    });

    it('Check that loading present, when submit and all is need to be disabled are disabled', done => {
        const wrapper = mount(<LogInScreen />);

        wrapper.setState({ username: 'asd', password: 'asdasd'});
        wrapper.find('#login').simulate('click', {preventDefault: () => {}});
        wrapper.find('button').forEach((node, index) => expect(node.prop('disabled')).toEqual(true));
        wrapper.find('input').forEach((node, index) => expect(node.prop('disabled')).toEqual(true));
        expect(wrapper.state().loading).toBe(true);
        // expect(wrapper.find('Loading').length).toEqual(1);
console.log(renderer.create(<LogInScreen />).toJSON())
        done();
    });

});