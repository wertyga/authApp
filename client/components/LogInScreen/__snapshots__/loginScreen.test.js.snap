// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#LoginScreen component Check that component didn't change while require fields are empty 1`] = `
<div
  className="LogInScreen"
>
  <div
    className="form"
  >
    <label
      htmlFor="usernameInput"
    >
      Username:
    </label>
    <input
      disabled={false}
      id="usernameInput"
      name="username"
      onChange={[Function]}
      placeholder="Username..."
      type="text"
      value=""
    />
    <label
      htmlFor="passwordInput"
    >
      Password:
    </label>
    <input
      disabled={false}
      id="passwordInput"
      name="password"
      onChange={[Function]}
      placeholder="Password..."
      type="text"
      value=""
    />
    <button
      className="primary"
      disabled={true}
      id="login"
      onClick={[Function]}
    >
      Login
    </button>
    <button
      className="primary"
      disabled={false}
      id="registration"
      onClick={[Function]}
    >
      Registration
    </button>
  </div>
</div>
`;
